### ISSUE TITLE
Support different models for cleaning and final training in CleanLearning()

### ISSUE THREAD

**Issue by cgnorthcutt** [cleanlab/cleanlab#82]
Hi @daMichaelB . This is a great suggestion. I think this is a great feature to add and we'll keep this in mind. One trick you can use for now is to obtain the label errors yourself via `cleanlab.pruning.get_noise_indices()` using a smaller model trained with fewer epochs. But note that if you use a significantly worse model, it will affect the quality of the errors found. As a rule of thumb, I recommend training a model within at least 5% accuracy of your fully-trained, top performing model. 

In response to your original question, you can use packages like https://github.com/skorch-dev/skorch to auto-integrate pytorch models into cleanlab without any custom code.

_Originally posted by @cgnorthcutt in https://github.com/cleanlab/cleanlab/discussions/69#discussioncomment-1369861_

**Comment by jwmueller** [comment]
A special instance of this use-case is to use the same classifier but with two sets of hyperparameters:  https://github.com/cleanlab/cleanlab/issues/82

**Comment by jwmueller** [comment]
This is already supported. Here's how to use one type of model for finding label issues and another type of model for the final training on the clean subset of data with label issues removed:

```
from cleanlab.classification import CleanLearning
import numpy as np
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import GradientBoostingClassifier

# Make data
data = np.vstack([np.random.random((100, 2)), np.random.random((100, 2)) + 10])
y = np.array([0] * 100 + [1] * 100)
# Introduce label errors
true_errors = [97, 98, 100, 101, 102, 104]
for idx in true_errors:
	y[idx] = 1 - y[idx]  # Flip label

# Demonstrate CleanLearning with 2 different classifiers:

model_to_find_errors = LogisticRegression()  # this model will be trained many times  via cross-validation
model_to_return = GradientBoostingClassifier()  # this model will be trained once on clean subset of data
cl0 = CleanLearning(model_to_find_errors)
issues = cl0.find_label_issues(data, y)
print(cl0.clf)  # will be LogisticRegression()

cl = CleanLearning(model_to_return).fit(data, y, label_issues=issues)
pred_probs = cl.predict_proba(data)  # predictions from GradientBoostingClassifier
print(cl.clf)  # will be GradientBoostingClassifier()
```

**Comment by jwmueller** [comment]
This strategy can be especially useful for hyperparameter-tuning the final model returned by CleanLearning, but not utilizing hyperparameter-tuning to find the label issues, which could otherwise be much more expensive due to the cross-validation step.  Here's an example of this in action:

```
import numpy as np
from cleanlab.classification import CleanLearning
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.model_selection import RandomizedSearchCV

# Make data
data = np.vstack([np.random.random((100, 2)), np.random.random((100, 2)) + 10])
y = np.array([0] * 100 + [1] * 100)
# Introduce label errors
true_errors = [97, 98, 100, 101, 102, 104]
for idx in true_errors:
	y[idx] = 1 - y[idx]  # Flip label

# Demonstrate CleanLearning with no hyperparameter-tuning to find label issues
# but hyperparameter-tuning for the final training of model on clean subset of the data:
model_to_find_errors = GradientBoostingClassifier()  # this model will be trained many times  via cross-validation
model_to_return = RandomizedSearchCV(GradientBoostingClassifier(),
 				      param_distributions = {
 				          "learning_rate": [0.001, 0.05, 0.1, 0.2, 0.5],
 				          "max_depth": [3, 5, 10],
 				      }
				  )   # this model will be trained once on clean subset of data
cl0 = CleanLearning(model_to_find_errors)
issues = cl0.find_label_issues(data, y)

cl = CleanLearning(model_to_return).fit(data, y, label_issues=issues)
pred_probs = cl.predict_proba(data)  # predictions from hyperparameter-tuned GBDT
```

**Comment by jwmueller** [comment]
More comprehensive answers to such questions have been added to our new FAQ:
https://docs.cleanlab.ai/master/tutorials/faq.html#How-can-I-use-different-models-for-data-cleaning-vs.%C2%A0final-training-in-CleanLearning?
